setup_chef_server.sh

```diff
@@ -36,25 +39,51 @@ pushd /home/vagrant/chef-bcpc/lib/cluster-def-gem > /dev/null
 sudo /opt/chefdk/embedded/bin/gem install cluster_def
 popd > /dev/null
 
-if dpkg -s chef-server 2>/dev/null | grep -q ^Status.*installed; then 
-# Faraday will be required in libs, so we will not have a chance to recipe this away
+if dpkg -s chef-server-core 2>/dev/null | grep -q ^Status.*installed; then
   chef-server-ctl restart
   echo 'chef-server is installed and the server has been restarted'
 else
-  apt-get -y install chef-server
+  apt-get -y install chef-server-core
   mkdir -p /etc/chef-server
-  printf "chef_server_webui['enable'] = false\n" >> /etc/chef-server/chef-server.rb
-  printf "nginx['enable_non_ssl'] = false\n" >> /etc/chef-server/chef-server.rb
-  printf "nginx['non_ssl_port'] = 4000\n" >> /etc/chef-server/chef-server.rb
-  # Configure Solr to index right away when we a new node.  
-  # Reference: https://docs.chef.io/config_rb_server.html#opscode-solr4
-  # Called opscode_solr4 in chef-server 12+
-  printf "chef_solr['max_commit_docs'] = 1\n" >> /etc/chef-server/chef-server.rb
+  printf "nginx['enable_non_ssl'] = false\n" >> /etc/opscode/chef-server.rb
+  printf "nginx['non_ssl_port'] = 4000\n" >> /etc/opscode/chef-server.rb
   # we can take about 45 minutes to Chef the first machine when running on VMs
   # so follow tuning from CHEF-4253
-  printf "erchef['s3_url_ttl'] = 3600\n" >> /etc/chef-server/chef-server.rb
-  export NO_PROXY=${NO_PROXY-127.0.0.1}
+  printf "opscode_erchef['s3_url_ttl'] = 3600\n" >> /etc/opscode/chef-server.rb
+  export NO_PROXY=${NO_PROXY-127.0.0.1,$(hostname),$(hostname -f)}
   chef-server-ctl reconfigure
+
+  # Create Chef Admin User
```

Comment:
We currently have the bootstrap as the "admin" user. Why are we using the pivotal user in the chef-server as the admin for the org?

If we want a separate admin account we should recreate it. So the bootstrap will now have 3 certs:


1.  pivotal (ishould be optional) - for the `chef-server-ctl` commands. we shouldn't touch this so we can move to a chef-server we don't manage someday.
2.  bootstrap node's client.pem - to converge the bootstrap node.
3.  bootstrap's knife client.pem - to do "admin-ish" operations on the organization like create databags, assign runlists, etc.

Reply:
I certainly agree! This is the plumbing to get us to 12 but it does not do 12 right; this simply gets us to a point we can leverage a 12 server. Let us design the correct permissioning in #1240. Copied your points there.